src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -105,6 +106,29 @@ public Dataset find(Object pk) {
         return em.find(Dataset.class, pk);
     }
 
+    public Dataset findDeep(Object pk) {
+        return (Dataset) em.createNamedQuery("Dataset.findById")
+            .setParameter("id", pk)
+            .setHint("eclipselink.left-join-fetch", "o.files.ingestRequest")
+            .setHint("eclipselink.left-join-fetch", "o.files.thumbnailForDataset")
+            .setHint("eclipselink.left-join-fetch", "o.files.dataTables")
+            .setHint("eclipselink.left-join-fetch", "o.files.auxiliaryFiles")
+            .setHint("eclipselink.left-join-fetch", "o.files.ingestReports")
+            .setHint("eclipselink.left-join-fetch", "o.files.dataFileTags")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileMetadatas")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileMetadatas.fileCategories")
+            .setHint("eclipselink.left-join-fetch", "o.files.guestbookResponses")
+            .setHint("eclipselink.left-join-fetch", "o.files.embargo")
+            .setHint("eclipselink.left-join-fetch", "o.files.fileAccessRequests")
+            .setHint("eclipselink.left-join-fetch", "o.files.owner")
+            .setHint("eclipselink.left-join-fetch", "o.files.releaseUser")
+            .setHint("eclipselink.left-join-fetch", "o.files.creator")
+            .setHint("eclipselink.left-join-fetch", "o.files.alternativePersistentIndentifiers")
+            .setHint("eclipselink.left-join-fetch", "o.files.roleAssignments")
+            .setLockMode(LockModeType.NONE)
```

Comment:
When is this OK? If I understand, in the indexing case this makes sense because a) you're not expecting any writes and b) if there are some, you're going to reindex anyway. Assuming this is saving significant time, can you add a comment about when this shouldn't be used. (Common theme through a few of my comments - trying to making sure that developers who don't have deep JPA/SQL knowledge or who just see the method name somewhere don't misuse them.)

Reply:
I have added comments to that method. As for the lock mode, this is just explicit default ostrich locking (default for em.find() and named query executions). I made it explicit to think about it later. I think this is always OK when reading and writing in a single transaction. Otherwise, you can have behind-your-back modification problems, which should not be common in the Dataverse context, since we do not expect multiple users working on the same dataset simultaneously.
I have removed the setLockMode line just in case when there could be a side-effect I am not aware of. Also, this should prevent any confusion about it.